src/main/scala/firrtl/analyses/InstanceKeyGraph.scala

```diff
@@ -52,8 +85,33 @@ class InstanceKeyGraph(c: ir.Circuit) {
     */
   def findInstancesInHierarchy(module: String): Seq[Seq[InstanceKey]] = {
     val instances = vertices.filter(_.module == module).toSeq
-    instances.flatMap{ i => fullHierarchy.getOrElse(i, Nil) }
+    instances.flatMap{ i => cachedFullHierarchy.getOrElse(i, Nil) }
   }
+
+  /** Given a circuit, returns a map from module name to a map
+    * in turn mapping instances names to corresponding module names
+    */
+  def getChildInstanceMap: mutable.LinkedHashMap[OfModule, mutable.LinkedHashMap[Instance, OfModule]] =
+    mutable.LinkedHashMap(childInstances.map { case (k, v) =>
+      val moduleMap: mutable.LinkedHashMap[Instance, OfModule] = mutable.LinkedHashMap(v.map(_.toTokens):_*)
+      TargetToken.OfModule(k) -> moduleMap
+    }:_*)
+
+  /** All modules in the circuit reachable from the top module */
+  def reachableModules: Seq[OfModule] = cachedReachableModules
+
+  /** All modules *not* reachable from the top module of the circuit */
+  def unreachableModules: Seq[OfModule] = cachedUnreachableModules
+
+  /** An [[firrtl.graph.EulerTour EulerTour]] representation of the [[firrtl.graph.DiGraph DiGraph]] */
+  private lazy val tour = EulerTour(internalGraph, circuitTopInstance)
+
+  /** Finds the lowest common ancestor instances for two module names in a design */
+  def lowestCommonAncestor(moduleA: Seq[InstanceKey], moduleB: Seq[InstanceKey]): Seq[InstanceKey] =
+    tour.rmq(moduleA, moduleB)
```

Comment:
I get the value of `InstanceKey` and that this is emulating the API from `InstanceGraph`, just wrapping calls to the `EulerTour`, but this is effectively the same thing as `InstanceTarget` so maybe it's better to share the API there?

Reply:
I moved the `lowestCommonAncestor` method out of `InstanceKeyGraph` and into the one place where is was used. Thus we do not add unnecessary API cruft.